runtime/engine/src/main/java/io/aklivity/zilla/runtime/engine/internal/stream/Target.java

```diff
@@ -299,6 +308,11 @@ private boolean handleWriteReply(
                 handled = streamsBuffer.test(msgTypeId, buffer, index, length);
                 streams[streamIndex(streamId)].remove(instanceId(streamId));
                 correlations.remove(streamId);
+                LongHashSet streamIdSet = streamSets.get(routeId);
+                if (streamIdSet != null)
+                {
+                    streamIdSet.remove(streamId);
+                }
```

Comment:
Can `streamIdSet` actually be `null`?

Reply:
Yes, it's needed, because when detaching a stream during reconfiguration, the DispatchAgent already removes the streamIdSet before triggering doSyntheticAbort/reset (which triggers these calls). This removal is only needed for the other syntheticAbort/reset calls, when closing Zilla regularly.